File: apps/demo-be-java-vertx/src/main/java/com/organiclever/demojavx/auth/PasswordService.java
package com.organiclever.demojavx.auth;

import org.mindrot.jbcrypt.BCrypt;

public class PasswordService {

    private static final int LOG_ROUNDS = 12;

    public String hash(String plaintext) {
        return BCrypt.hashpw(plaintext, BCrypt.gensalt(LOG_ROUNDS));
    }

    public boolean verify(String plaintext, String hashed) {
        return BCrypt.checkpw(plaintext, hashed);
    }
}

File: apps/crud-be-kotlin-ktor/src/main/kotlin/com/demobektkt/auth/PasswordService.kt
package com.demobektkt.auth

import org.mindrot.jbcrypt.BCrypt

class PasswordService {
  fun hash(plaintext: String): String = BCrypt.hashpw(plaintext, BCrypt.gensalt())

  fun verify(plaintext: String, hash: String): Boolean =
    runCatching { BCrypt.checkpw(plaintext, hash) }.getOrDefault(false)
}
